src/test/java/de/rabea/game/BoardTest.java

```diff
@@ -20,176 +20,298 @@ public void setup() {
         assertEquals(emptyCells, board.emptyCells());
     }
 
     @Test
-    public void indexOfLastCellInEachRow() {
-        Board board = new Board();
+    public void listsAllEmptyPositionsFor4x4() {
+        Board board = new Board(4);
+        board.placeMarkOnExistingBoard(1, X);
+        board.placeMarkOnExistingBoard(2, O);
+        board.placeMarkOnExistingBoard(3, X);
+        board.placeMarkOnExistingBoard(6, O);
+        List<Integer> emptyCells = Arrays.asList(0, 4, 5, 7, 8, 9, 10, 11, 12, 13, 14, 15);
+        assertEquals(emptyCells, board.emptyCells());
+    }
+
+    @Test
+    public void indexOfLastCellInEachRowOn3x3() {
+        Board board = new Board(3);
         List<Integer> cells = new ArrayList<>(Arrays.asList(2, 5, 8));
         assertEquals(cells, board.indexOfLastCellPerRow());
     }
 
     @Test
-    public void indexOfLastCellOfBoard() {
-        Board board = new Board();
+    public void indexOfLastCellOfBoardOn3x3() {
+        Board board = new Board(3);
         assertEquals(8, board.indexOfLastCell());
     }
 
     @Test
-    public void widthAndHeightOfTheBoard() {
-        Board board = new Board();
+    public void widthAndHeightOf3x3() {
+        Board board = new Board(3);
         assertEquals(3, board.getDimension());
     }
 
     @Test
-    public void isEndOfLastRow() {
-        Board board = new Board();
+    public void isEndOfLastRowFor3x3() {
+        Board board = new Board(3);
         assertTrue(board.isIndexOfLastCell(8));
         assertFalse(board.isIndexOfLastCell(7));
     }
 
     @Test
-    public void isEndOfFirstOrSecondRow() {
-        Board board = new Board();
-        assertTrue(board.isIndexOfEndOfFirstOrSecondRow(2));
-        assertTrue(board.isIndexOfEndOfFirstOrSecondRow(5));
-        assertFalse(board.isIndexOfEndOfFirstOrSecondRow(1));
-        assertFalse(board.isIndexOfEndOfFirstOrSecondRow(8));
+    public void isEndOfLastRowFor4x4() {
+        Board board = new Board(4);
+        assertTrue(board.isIndexOfLastCell(15));
+        assertFalse(board.isIndexOfLastCell(7));
     }
 
     @Test
-    public void gameOverWithoutWinner() {
-        Board board = fullBoardNoWinner();
+    public void isEndOfFirstOrSecondRowOn3x3() {
+        Board board = new Board(3);
+        assertTrue(board.isEndOfRowIndexExceptLastRow(2));
+        assertTrue(board.isEndOfRowIndexExceptLastRow(5));
+        assertFalse(board.isEndOfRowIndexExceptLastRow(1));
+        assertFalse(board.isEndOfRowIndexExceptLastRow(8));
+    }
+
+    @Test
+    public void isEndOfFirstOrSecondOrThirdRowOn4x4() {
+        Board board = new Board(4);
+        assertTrue(board.isEndOfRowIndexExceptLastRow(3));
+        assertTrue(board.isEndOfRowIndexExceptLastRow(7));
+        assertTrue(board.isEndOfRowIndexExceptLastRow(11));
+        assertFalse(board.isEndOfRowIndexExceptLastRow(15));
+    }
+
+    @Test
+    public void getsAllLinesOf4x4() {
```

Comment:
It helped me with updating the methods to get rows/columns/diagonals for a 4x4 board. 

I understand what you mean though. Maybe this is a test that is good for development but can be deleted once the production code works as expected. The methods are still tested through other tests.